### DISCUSSION TITLE
When adding a routed event handler, I can't get the necessary control.

### DISCUSSION THREAD

**Discussion by FibonacciFox** [AvaloniaUI/Avalonia#14496] [Q&A]
![image](https://github.com/AvaloniaUI/Avalonia/assets/61143434/e2ead98e-957f-4fea-89e3-340f0f5d7244)

To get control when the mouse is clicked, I use the following code :
```c#
_targetControl.AddHandler(PointerPressedEvent, (sender, e) =>
        {
            var control = e.Source as Control;
            Console.WriteLine(control?.Name);
            
            e.Handled = true;
        }, RoutingStrategies.Tunnel | RoutingStrategies.Bubble); 

_targetControl.AddHandler(TappedEvent, (sender, e) =>
        {

            var control = sender as Control;

            Console.WriteLine(control?.Name);
            
            e.Handled = true;
        }, RoutingStrategies.Tunnel | RoutingStrategies.Bubble);
```       

**When I click on the control itself, I only get the name of the panels correctly. Although, if I use Tapped Event, then I can get any control in this way.**

_MyStackPanel
MyPanel
PART_TextPresenter
PART_TextPresenter
PART_ContentPresenter
PART_BorderElement_

https://github.com/FibonacciFox/DesignerPanel/blob/master/Decorator.axaml.cs#L45-L64

https://github.com/FibonacciFox/DesignerPanel/tree/master

**Comment by timunie** [comment]
You probably need to use the VisualTreeExtensions to find the parent of e.Source

**Reply by FibonacciFox** [reply]
The problem is that I can't figure out how to do this using VisualTreeExtensions.

**Reply by timunie** [reply]
`myControl.FindAncestorOfType<Button>()`

**Reply by FibonacciFox** [reply]
Controls can be of any type.
Sorting through all types will be very expensive, and custom controls will also have to be added.

**Comment by FibonacciFox** [comment]
The solution to the problem could not be found

**Reply by IanRawley** [reply]
Do you want the control that the handler has been added to?  If so, use sender rather than e.source.

**Reply by FibonacciFox** [reply]
Alas, this does not solve the problem, a link to the repository is attached
